src/VariationPriceHandler.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Drupal\commerce_attributes;
+
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\commerce_price\Price;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\commerce_product\ProductAttributeFieldManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\commerce_attributes\Entity\ProductAttributeValue;
+use Drupal\commerce_attributes\Entity\ProductAttribute;
+
+/**
+ * Handles the Variation price calculations.
+ */
+class VariationPriceHandler {
+
+  /**
+   * Logger Service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $Logger;
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * The attribute field manager.
+   *
+   * @var \Drupal\commerce_product\ProductAttributeFieldManagerInterface
+   */
+  protected $attributeFieldManager;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(LoggerChannelFactoryInterface $logger_factory, EntityManagerInterface $entity_manager, EntityTypeManagerInterface $entity_type_manager, ProductAttributeFieldManagerInterface $attribute_field_manager) {
+    $this->logger = $logger_factory->get('system');
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityManager = $entity_manager;
+    $this->attributeFieldManager = $attribute_field_manager;
+  }
+
+  /**
+   * Retrieves the product variations.
+   *
+   * @param array $types
+   *   The product variation types.
+   * @param string $attribute
+   *   The attribute machine name.
+   * @param int $attributeValue
+   *   The Attribue value id.
+   *
+   * @return array
+   *   Array of variations id.
+   */
+  public function getProductVariation(array $types, $attribute, $attributeValue) {
+    // The attribute field name of variation type will always be in the
+    // format of `attrubute_attribute_machine_name`.
+    $attributefieldName = 'attribute_' . $attribute;
+
+    // Get all variations associated with an particular attribute value.
+    $query = \Drupal::entityQuery('commerce_product_variation')
+      ->condition('type', $types, 'IN')
+      ->condition($attributefieldName, $attributeValue);
+    $variations = $query->execute();
+    return $variations;
+  }
```

Comment:
**CHANGE REQUEST**

* You're not getting one variation here, you get multiple; update the method name accordingly.
* No `Drupal::` in classes please. Add an extra property in the class `$variationStorage` and get it in the constructor from the entity type manager. You will then be able to use it in other functions as well.
* Many unused variables here - you increase memory consumption for no reason at all.

```
protected function getVariations(array $types, $attribute, $attribute_value) {
    // The attribute field name of variation type will always be in the
    // format of `attrubute_attribute_machine_name`.
    $attribute_field_name = 'attribute_' . $attribute;

    // Get all variations associated with an particular attribute value.
    return $this->variationStorage
      ->condition('type', $types, 'IN')
      ->condition($attribute_field_name, $attribute_value)
      ->execute();
  }
```

Reply:
Done